UI/MapWnd.cpp

```diff
@@ -4065,9 +4060,13 @@ void MapWnd::InitFieldRenderingBuffers() {
         if (!field_texture)
             continue;
 
-        auto& field_both_vertex_buffers = m_field_vertices[field_texture];
+        // group by texture as much as possible for fewer GL calls, but generally paint fields one by one according to size
+        // so smaller ones get painted over larger ones, including across different textures
+        auto& field_vertices = current_field_visible ? m_field_vertices_visible : m_field_vertices_not_visible;
         GG::GL2DVertexBuffer& current_field_vertex_buffer =
-            current_field_visible ? field_both_vertex_buffers.first : field_both_vertex_buffers.second;
+            (field_vertices.empty() || field_vertices.back().first != field_texture) ?
```

Comment:
this is a bit hard to parse...

Reply:
umm should I put some comment on it then or...? I kept it referring to ``GG::GL2DVertexBuffer& current_field_vertex_buffer `` so there are fewer changes between versions, showing parts did not change. 

The condition means if the buffer collection is empty, or last "buffered" texture is different than the current one, we need to create new vertex buffer for current texture and add new vertices to that, otherwise we can keep adding to the old vertex buffer, because texture didn't change. This groups the gl calls per texture as much as possible while still outputting fields in order of size accross textures

Reply:
maybe just put the `(field_vertices.empty() || field_vertices.back().first != field_texture)` into a named variable instead of in the same big expression as the `emplace` call?

Reply:
ok, added a further comment and a named bool `should_create_new_buffer`